modules/standard/Sets.chpl

```diff
@@ -0,0 +1,496 @@
+  // list, or any operation that is O(n)).
+  //
+  pragma "no doc"
+  type _lockType = ChapelLocks.chpl_LocalSpinlock;
+
+  //
+  // Use a wrapper class to let list methods have a const ref receiver even
+  // when `parSafe` is `true` and the list lock is used.
+  //
+  pragma "no doc"
+  class _LockWrapper {
+    var lock$ = new _lockType();
+
+    inline proc lock() {
+      lock$.lock();
+    }
+
+    inline proc unlock() {
+      lock$.unlock();
+    }
+  }
+
+  /*
+    A set is a...
+  */
+  record set {
+
+    /* The type of the elements contained in this set. */
+    type eltType;
+
+    /* If `true`, this set will perform parallel safe operations. */
+    param parSafe = false;
+
+    pragma "no doc"
+    var _size = 0;
+
+    pragma "no doc"
+    var _lock$ = if parSafe then new _LockWrapper() else none;
+
+    /*
+      Initializes an empty set containing elements of the given type.
+
+      :arg eltType: The type of the elements of this set.
+      :arg parSafe: If `true`, this set will use parallel safe operations.
+    */
+    proc init(type eltType, param parSafe=false) {
+    }
+
+    /*
+      Initializes a set containing elements that are copy initialized from
+      the elements contained in another set.
+
+      :arg other: The set to initialize from.
+      :arg parSafe: If `true`, this set will use parallel safe operations.
+    */
+    proc init=(const ref other: set(?t), param parSafe=other.parSafe) {
+    }
+
+    pragma "no doc"
+    inline proc deinit() {
+    }
+
+    pragma "no doc"
+    inline proc _enter() {
+      if parSafe then
+        _lock$.lock();
+    }
+
+    pragma "no doc"
+    inline proc _leave() {
+      if parSafe then
+        _lock$.unlock();
+    }
+
+    /*
+      Add a copy of the element `x` to this set. Does nothing if this set
+      already contains an element equal to the value of `x`.
+
+      :arg x: The element to add to this set.
+    */
+    proc add(const ref x: eltType) {
+      return;
+    }
+
+    /*
+      Returns `true` if the given element is a member of this set, and `false`
+      otherwise.
+
+      :arg x: The element to test for membership.
+      :return: Whether or not the given element is a member of this set.
+    */
+    proc const contains(const ref x: eltType): bool {
+      return false;
+    }
+
+    /*
+      Returns `true` if this set shares no elements in common with the set
+      `other`, and `false` otherwise.
+
+      :arg other: The set to compare against.
+      :return: Whether or not this set and `other` are disjoint.
+    */
+    proc const disjoint(const ref other: set(eltType, ?)): bool {
+      return false;
+    }
+
+    /*
+      Returns `true` if this set and `other` have at least one element in
+      common.
+
+      :arg other: The set to compare against.
+      :return: Whether or not this set and `other` intersect.
+    */
+    proc const intersect(const ref other: set(eltType, ?)): bool {
```

Comment:
I think `intersect` is already a verb and it means to compute the intersection of the set with another set. That would result in a set. However this returns a bool. How about naming it `isIntersecting` ?